cmd/gather-sysinfo/gather-sysinfo.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"fmt"
+	"github.com/openshift-kni/debug-tools/pkg/machineinformer"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/jaypipes/ghw/pkg/snapshot"
+	"github.com/openshift-kni/debug-tools/pkg/knit/cmd"
+	"github.com/spf13/cobra"
+)
+
+const machineInfoFilePath string = "machineinfo.json"
+
+type snapshotOptions struct {
+	dumpList  bool
+	output    string
+	rootDir   string
+	sleepTime int
+}
+
+func main() {
+	root := cmd.NewRootCommand(newSnapshotCommand)
+
+	if err := root.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func newSnapshotCommand(knitOpts *cmd.KnitOptions) *cobra.Command {
+	opts := &snapshotOptions{}
+	snap := &cobra.Command{
+		Use:   "snapshot",
+		Short: "snapshot pseudofilesystems for offline analysis",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return makeSnapshot(cmd, knitOpts, opts, args)
+		},
+		Args: cobra.NoArgs,
+	}
+	snap.Flags().StringVar(&opts.rootDir, "root", "", "pseudofs root - use this if running inside a container")
+	snap.Flags().StringVar(&opts.output, "output", "", "path to clone system information into")
+	snap.Flags().BoolVar(&opts.dumpList, "dump", false, "just dump the glob list of expected content and exit")
+	// use this to debug container behaviour
+	snap.Flags().IntVar(&opts.sleepTime, "sleep", 0, "amount of seconds to sleep once done, before exit")
```

Comment:
this raised concerns in review and we can probably replace anyway with something better. I'd remove it.